conda/recipes/cudf/meta.yaml

```
@@ -19,44 +19,46 @@ build:
   script_env:
     - VERSION_SUFFIX
     - PARALLEL_LEVEL
-    - CC
-    - CXX
-    - CUDAHOSTCXX
   # libcudf's run_exports pinning is looser than we would like
   ignore_run_exports:
     - libcudf
 
 requirements:
   build:
-    - protobuf
-    - python
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - sysroot_linux-64 {{ sysroot_version }}  # [linux64]
+    - sysroot_linux-aarch64 {{ sysroot_version }}  # [aarch64]
```

Do we mean the following? I don't know if `sysroot_linux-64` is only supposed to be installed on `x86_64`. See [preprocessing selectors](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors). I believe that Linux ARM systems match both `linux64` and `aarch64`.
```suggestion
    - sysroot_linux-64 {{ sysroot_version }}  # [linux64 and x86_64]
    - sysroot_linux-aarch64 {{ sysroot_version }}  # [linux64 and aarch64]
```

---

It is

---

I don't think that's the case, but I'm not able to find the conda source code that handles this. I think this pattern is misused by a lot of recipes and feedstocks which default to `linux64` and overwrite the value on `aarch64` and other platforms, if my understanding of selector logic is correct. However, this isn't a case where we're overriding a single value, but adding multiple values to the same list. If the recipe works as-is, we don't need to block on this, but I would like to clarify my understanding since "`linux64` is for `x86_64` only" does not align with the docs I linked above. cc: @jakirkham in case you have insight here.

---

Sorry @jakirkham I missed your comment. I see it now that I refreshed. All good.

---

That said, we can simplify this like so

```suggestion
    - sysroot_{{ target_platform }} {{ sysroot_version }}
```

---

Also submitted PR ( https://github.com/conda/conda-build/pull/4476 ) to clarify the docs